plugins/ftdidmx/FtdiDmxThread.cpp

```diff
@@ -73,15 +97,133 @@ bool FtdiDmxThread::WriteDMX(const DmxBuffer &buffer) {
   }
 }
 
+void FtdiDmxThread::SendRDMRequest(ola::rdm::RDMRequest *request,
+                    ola::rdm::RDMCallback *callback) {
+  ola::thread::MutexLocker locker(&m_rdm_mutex);
+  if(m_pending_request == nullptr) {
+      m_pending_request = request;
+      m_rdm_callback = callback;
+  } else {
+    OLA_WARN << "Unable to queue RDM request, RDM operation already pending";
+  }
+  OLA_WARN << "Function not properly implemented yet, callback may not get called.";
+}
+
+void FtdiDmxThread::RunFullDiscovery(ola::rdm::RDMDiscoveryCallback *callback) {
+  m_discovery_agent.StartFullDiscovery(ola::NewSingleCallback(this, &FtdiDmxThread::DiscoveryComplete, callback));
+}
+
+void FtdiDmxThread::RunIncrementalDiscovery(ola::rdm::RDMDiscoveryCallback *callback) {
+  m_discovery_agent.StartIncrementalDiscovery(ola::NewSingleCallback(this, &FtdiDmxThread::DiscoveryComplete, callback));
+}
+
+/**
+ * Called when the discovery process finally completes
+ * @param callback the callback passed to StartFullDiscovery or
+ * StartIncrementalDiscovery that we should execute.
+ * @param status true if discovery worked, false otherwise
+ * @param uids the UIDSet of UIDs that were found.
+ */
+void FtdiDmxThread::DiscoveryComplete(ola::rdm::RDMDiscoveryCallback *callback,
+                                       bool,
```

Comment:
Can you switch to OLA_UNUSED

Reply:
This is an exact copy from EnttecUSB
It seems that all the DiscoveryComplete functions completely ignore the status bool.

Either way I have added minor debug/warn output.

Reply:
Don't feel you need to use it for some debug if it's ignored elsewhere. I suspect nothing bothers logging as the discovery controller will have logged the same error. It was more about moving from (void) to OLA_UNUSED to better show we don't process it. The Enttec code will get updated in due course when it next gets touched.

Reply:
still not sure what you are referring to here?